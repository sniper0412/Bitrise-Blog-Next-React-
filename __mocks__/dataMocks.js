export const mockAuthors = [
  {
    slug: 'author-1',
    firstName: 'Bender',
    lastName: 'Rodr√≠guez',
    profileImage: 'http://bender.img',
    title: 'Chief Bending Robot',
    bio: 'A fictional character who is one of the main characters in the animated television series Futurama'
  }
];

export const mockCategories = [
  {
    slug: 'category-1',
    name: 'category 1'
  },
  {
    slug: 'category-2',
    name: 'category 2'
  }
];

export const mockTags = [
  {
    slug: 'tag-1',
    name: 'tag 1'
  },
  {
    slug: 'tag-2',
    name: 'tag 2'
  }
];

export const mockPosts = [
  {
    author: mockAuthors[0],
    slug: 'posts-1',
    seoTitle: 'post 1',
    metaDescription: 'Post 1 description',
    featuredImage: 'http://post-1.img',
    title: 'Post 1',
    body: '<h1>Post 1 body</h1>',
    categories: [mockCategories[0], mockCategories[1]],
    tags: [mockTags[0], mockTags[1]],
    published: 'Wed Jan 02 2019 16:50:20 GMT+0100',
    summary: 'Post 1 summary'
  }
];
