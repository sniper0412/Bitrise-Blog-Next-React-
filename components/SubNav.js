import React from 'react';
import SVG from 'react-inlinesvg';
import cx from 'classnames';

import { ROUTE_PATHS } from '../config';

const Category = ({ data: { slug, name } }) => (
  <a href={`${ROUTE_PATHS.categories}/${slug}`} title={name} className="category">
    {name}
  </a>
);

export default class SubNav extends React.Component {
  state = {
    isDropDownOpen: false
  };

  onSelectCategory() {
    this.setState({ isDropDownOpen: !this.state.isDropDownOpen });
  }

  static defaultProps = {
    categories: []
  };

  render() {
    const { categories } = this.props;
    const { isDropDownOpen } = this.state;

    const searchForm = (
      <form action="/search" className="search-wrapper">
        <SVG src="/static/svg/search.svg" />
        <input name="q" type="text" placeholder="Search articles" />
      </form>
    );

    return (
      <div className="categories">
        <div className="category-select-wrapper" onClick={() => this.onSelectCategory()}>
          <SVG src="/static/svg/filter.svg" />
          <button className="category-select-mobile category-select">filter by category</button>
          <div className={cx('dropdown-arrow-container', isDropDownOpen ? 'opened' : 'closed')}>
            <SVG src="/static/svg/dropdown_arrow.svg" />
          </div>
        </div>
        <div className={cx('content-wrapper', { opened: isDropDownOpen })}>
          <div className="search-mobile">{searchForm}</div>
          <a href="/" title="Home" className="category home">
            <SVG src="/static/svg/home_button.svg" />
            <span>Home</span>
          </a>
          {categories.map(category => (
            <Category data={category} key={category.slug} />
          ))}
          <div className="search-container desktop">{searchForm}</div>
        </div>
      </div>
    );
  }
}
