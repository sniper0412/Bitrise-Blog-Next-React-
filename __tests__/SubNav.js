import renderer from 'react-test-renderer';

import { mockCategories } from 'dataMocks';
import SubNav from '../components/SubNav';

describe('SubNav component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SubNav categories={mockCategories} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
