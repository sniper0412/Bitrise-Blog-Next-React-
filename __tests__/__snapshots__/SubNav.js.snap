// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubNav component renders correctly 1`] = `
<div
  className="categories"
>
  <div
    className="category-select-wrapper"
    onClick={[Function]}
  >
    <span
      className="isvg loading"
    />
    <button
      className="category-select-mobile category-select"
    >
      filter by category
    </button>
    <div
      className="dropdown-arrow-container closed"
    >
      <span
        className="isvg loading"
      />
    </div>
  </div>
  <div
    className="content-wrapper"
  >
    <div
      className="search-mobile"
    >
      <div
        className="ais-InstantSearch__root"
      >
        <form
          action="/search"
          noValidate={true}
          role="search"
        >
          <span
            className="isvg loading"
          />
          <input
            autoComplete="off"
            name="q"
            onChange={[Function]}
            placeholder="Search articles"
            type="search"
            value=""
          />
        </form>
      </div>
    </div>
    <a
      className="category home"
      href="/"
      title="Home"
    >
      <span
        className="isvg loading"
      />
      <span>
        Home
      </span>
    </a>
    <a
      className="category"
      href="/categories/category-1"
      title="category 1"
    >
      category 1
    </a>
    <a
      className="category"
      href="/categories/category-2"
      title="category 2"
    >
      category 2
    </a>
    <div
      className="search-container desktop"
    >
      <div
        className="ais-InstantSearch__root"
      >
        <form
          action="/search"
          noValidate={true}
          role="search"
        >
          <span
            className="isvg loading"
          />
          <input
            autoComplete="off"
            name="q"
            onChange={[Function]}
            placeholder="Search articles"
            type="search"
            value=""
          />
        </form>
      </div>
    </div>
  </div>
</div>
`;
